polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -155,9 +158,72 @@ static bool isClassType(Type type, StringRef className) {
 }
 
 namespace {
-struct RaiseKernelName : public OpRewritePattern<LLVM::AddressOfOp> {
+template <typename SourceOp>
+struct OpOrInterfaceRaisePatternBase : public RewritePattern {
+  using RewritePattern::RewritePattern;
+
+  /// Wrappers around the RewritePattern methods that pass the derived op type.
+  void rewrite(Operation *op, PatternRewriter &rewriter) const final {
+    rewrite(cast<SourceOp>(op), rewriter);
+  }
+  LogicalResult match(Operation *op) const final {
+    return match(cast<SourceOp>(op));
+  }
```

Comment:
I might misunderstand the approach here, but don't you need the check for `GPUModuleOp` here as well?

Reply:
Nice catch. I wasn't catching this. Done!